packages/commons/src/interop-token/models.ts

```diff
@@ -3,134 +3,255 @@ import {
   ClientId,
   PurposeId,
   TenantId,
-  UserId,
 } from "pagopa-interop-models";
 import { z } from "zod";
-import { SystemRole } from "../auth/authData.js";
-
-export const ORGANIZATION = "organization";
-export const UID = "uid";
-export const NAME = "name";
-export const FAMILY_NAME = "family_name";
-export const EMAIL = "email";
-export const ORGANIZATION_ID_CLAIM = "organizationId";
-export const SELFCARE_ID_CLAIM = "selfcareId";
-export const ORGANIZATION_EXTERNAL_ID_CLAIM = "externalId";
-export const ORGANIZATION_EXTERNAL_ID_ORIGIN_CLAIM = "origin";
-export const ORGANIZATION_EXTERNAL_ID_VALUE_CLAIM = "value";
-export const USER_ROLES = "user-roles";
-const PURPOSE_ID_CLAIM = "purposeId";
-export const ROLE_CLAIM = "role";
-
-export interface InteropJwtHeader {
-  alg: string;
-  use: string;
-  typ: string;
-  kid: string;
-}
-
-export type InteropJwtCommonPayload = {
-  jti: string;
-  iss: string;
-  aud: string[] | string;
-  iat: number;
-  nbf: number;
-  exp: number;
-};
+import {
+  SUPPORT_USER_ID,
+  systemRole,
+  userRole,
+  UserRole,
+} from "../auth/roles.js";
+
+// Type utility to transform a string into an array of strings comma separated
```

Comment:
It does more than that 👇 

```suggestion
// Zod utility to parse a non-empty comma-separated string and transform it
// (e.g. `"foo,bar,baz"`) into an array whose elements are validated
// by the schema `t`.
//
// Example:
//   const NumberArray = CommaSeparatedStringToArray(z.number());
//   NumberArray.parse("1,2,3"); // → [1, 2, 3]
```